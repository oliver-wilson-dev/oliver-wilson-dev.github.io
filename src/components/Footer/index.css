.footerDisclaimerText {
  color: var(--primaryTextColorInverse);
  text-align: center;
  margin-bottom: 10px;

  &:last-child {
    margin-bottom: 0px;
  }
}

.container {
  margin-top: 30px;
  display: flex;
  padding: 10px 20px;
  font-size: 14px;
  justify-content: space-between;
  /* background-color: var(--primaryBackgroundLight); */
  background-color: var(--primaryBackgroundInverse);

  & a {
    color: var(--unvisitedLinkInverse);
  }

  * {
    text-align: left;
  }
}

.buttonLink {
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  cursor: pointer;
  color: var(--primaryTextColorInverse);
  text-decoration: underline;
}

.textContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}
