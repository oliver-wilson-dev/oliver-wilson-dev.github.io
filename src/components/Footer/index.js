import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import additionalStyles from '../App/index.css';
import styles from './index.css';
import SocialMedia from '../SocialMedia';


const Footer = ({ learnMoreClicked, sectionsContentFetched }) => sectionsContentFetched && (
  <div className={cn(additionalStyles.flexCenter, styles.container)}>
    <div className={styles.textContainer}>
      <span className={styles.footerDisclaimerText}>
        By using this site you agree to our cookie policy.
        {' '}
        <button className={cn(styles.buttonLink)} type="button" onClick={learnMoreClicked}>
          Read the cookie policy here
          <span>.</span>
        </button>
      </span>
      <span className={styles.footerDisclaimerText}>
        The code for this website can be found on
        {' '}
        <a href="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io" target="_blank" rel="noopener noreferrer">github</a>
        .
      </span>
      <span className={styles.footerDisclaimerText}>Â©2020, Oliver Wilson</span>
    </div>
    <SocialMedia />
  </div>
);

Footer.propTypes = {
  learnMoreClicked: PropTypes.func.isRequired,
  sectionsContentFetched: PropTypes.bool.isRequired
};

export default Footer;
