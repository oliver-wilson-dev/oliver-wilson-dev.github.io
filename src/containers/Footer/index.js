import { connect } from 'react-redux';
import Footer from '../../components/Footer';
import learnMoreClicked from '../../state/actions/learnMoreClicked';
import { getSectionsContentFetched } from '../../state/selectors';

const mapStateToProps = state => ({
  sectionsContentFetched: getSectionsContentFetched(state)
});
const mapDispatchToProps = dispatch => ({
  learnMoreClicked: () => dispatch(learnMoreClicked())
});

export default connect(mapStateToProps, mapDispatchToProps)(Footer);
