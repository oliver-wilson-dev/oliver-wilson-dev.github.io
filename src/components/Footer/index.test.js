import React from 'react';
import { shallow } from 'enzyme';
import Footer from '.';

jest.mock('../SocialMedia', () => {
  const SocialMedia = () => null;

  return SocialMedia;
});

const defaultProps = {
  learnMoreClicked: jest.fn(),
  sectionsContentFetched: true
};

const render = (props = {}) => shallow(<Footer {...defaultProps} {...props} />);

describe('<Footer/>', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when the learn more button is clicked', () => {
    it('should call the learnMoreClicked prop', () => {
      const component = render();

      component.find('button').simulate('click');

      expect(defaultProps.learnMoreClicked).toHaveBeenCalled();
    });
  });

  describe('when sectionsContentFetched is false', () => {
    it('should render null', () => {
      expect(render({ sectionsContentFetched: false })).toMatchSnapshot();
    });
  });
});
