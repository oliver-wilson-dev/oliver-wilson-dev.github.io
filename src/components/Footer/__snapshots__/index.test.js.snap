// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render correctly 1`] = `
<div
  className="flexCenter container"
>
  <div
    className="textContainer"
  >
    <span
      className="footerDisclaimerText"
    >
      By using this site you agree to our cookie policy.
       
      <button
        className="buttonLink"
        onClick={[MockFunction]}
        type="button"
      >
        Read the cookie policy here
      </button>
      .
    </span>
    <span
      className="footerDisclaimerText"
    >
      The code for this website can be found on
       
      <a
        href="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io"
        rel="noopener noreferrer"
        target="_blank"
      >
        github
      </a>
      .
    </span>
    <span
      className="footerDisclaimerText"
    >
      Â©2020, Oliver Wilson
    </span>
  </div>
  <SocialMedia />
</div>
`;
